docs/plugins/development/creating-text-analysis-plugins.asciidoc

```diff
@@ -0,0 +1,129 @@
+[[creating-text-analysis-plugins]]
+=== Creating text analysis plugins
+
+Text analysis plugins provide {es} with custom Lucene analyzers, token filters, 
+character filters, and tokenizers. 
```

Comment:
maybe we could link to https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis.html here?

Reply:
Good idea. Done.